src/collectors/mdstat/test/test_mdstat.py

```diff
@@ -0,0 +1,230 @@
+
+    @patch.object(Collector, 'publish')
+    def test_mdstat_multipath(self, publish_mock):
+        MdStatCollector.MDSTAT_PATH = self.getFixturePath('mdstat_multipath')
+        self.collector.collect()
+
+        metrics = {
+            'md0.status.superblock_version': 1.2,
+            'md0.status.actual_members': 2,
+            'md0.status.total_members': 2,
+            'md0.status.blocks': 102320,
+            'md0.member_count.active': 2,
+            'md0.member_count.faulty': 0,
+            'md0.member_count.spare': 0
+        }
+
+        self.assertPublishedMany(publish_mock, metrics)
+
+    @patch.object(Collector, 'publish')
+    def test_mdstat_raid1(self, publish_mock):
+        MdStatCollector.MDSTAT_PATH = self.getFixturePath('mdstat_raid1')
+        self.collector.collect()
+
+        metrics = {
+            'md0.status.superblock_version': 1.2,
+            'md0.status.actual_members': 2,
+            'md0.status.total_members': 2,
+            'md0.status.blocks': 100171776,
+            'md0.member_count.active': 2,
+            'md0.member_count.faulty': 0,
+            'md0.member_count.spare': 0,
+            'md0.bitmap.total_pages': 1,
+            'md0.bitmap.allocated_pages': 1,
+            'md0.bitmap.page_size': 4,
+            'md0.bitmap.chunk_size': 65536
+        }
+
+        self.assertPublishedMany(publish_mock, metrics)
+
+    @patch.object(Collector, 'publish')
+    def test_mdstat_raid1_failed(self, publish_mock):
+        MdStatCollector.MDSTAT_PATH = \
+            self.getFixturePath('mdstat_raid1-failed')
+        self.collector.collect()
+
+        metrics = {
+            'md0.status.superblock_version': 1.2,
+            'md0.status.actual_members': 1,
+            'md0.status.total_members': 2,
+            'md0.status.blocks': 102272,
+            'md0.member_count.active': 1,
+            'md0.member_count.faulty': 1,
+            'md0.member_count.spare': 0
+        }
+
+        self.assertPublishedMany(publish_mock, metrics)
+
+    @patch.object(Collector, 'publish')
+    def test_mdstat_raid1_recover(self, publish_mock):
+        MdStatCollector.MDSTAT_PATH = \
+            self.getFixturePath('mdstat_raid1-recover')
+        self.collector.collect()
+
+        metrics = {
+            'md0.status.superblock_version': 1.2,
+            'md0.status.actual_members': 1,
+            'md0.status.total_members': 2,
+            'md0.status.blocks': 102272,
+            'md0.recovery.percent': 99.5,
+            'md0.recovery.speed': 104726528,
+            'md0.recovery.remaining_time': 802199,
+            'md0.member_count.active': 2,
+            'md0.member_count.faulty': 0,
+            'md0.member_count.spare': 0,
+            'md0.bitmap.total_pages': 1,
+            'md0.bitmap.allocated_pages': 1,
+            'md0.bitmap.page_size': 4,
+            'md0.bitmap.chunk_size': 65536
+        }
+
+        self.setDocExample(collector=self.collector.__class__.__name__,
```

Comment:
`self.setDocExample`  should only be called once per test file, so one of the calls needs to be removed